tree/treeplayer/src/TTreeProcessorMT.cxx

```diff
@@ -562,8 +467,8 @@ void TTreeProcessorMT::Process(std::function<void(TTreeReader &)> func)
    // Per-file processing in case we retrieved all cluster info upfront
    auto processFileUsingGlobalClusters = [&](std::size_t fileIdx) {
       auto processCluster = [&](const EntryRange &c) {
-         auto r = fTreeView->GetTreeReader(c.first, c.second, fTreeNames, fFileNames, fFriendInfo, fEntryList,
-                                           allEntries, GetFriendEntries(fFriendInfo));
```

Comment:
Just wanted to point out that previously the call to `GetFriendEntries` was done inside the lambda that processes each `EntryRange` of a certain file. With this PR the gathering of the friend entries is moved at construction time of the TTreeProcessorMT. This in principle should mean cutting the number of times we call this function (thus opening all friend files) from NFILES*NENTRYRANGES to 1